tests/e2e/v3_curl_maxstream_test.go

```diff
@@ -162,22 +150,49 @@ func submitConcurrentWatch(cx ctlCtx, number int, wgDone *sync.WaitGroup, errCh
 	var wgSchedule sync.WaitGroup
 	wgSchedule.Add(number)
 	for i := 0; i < number; i++ {
-		go func(i int) {
+		go func(i int) (err0 error) {
+			defer func() {
+				if err0 != nil {
+					cx.t.Fatalf("testV3CurlMaxStream watch failed: %d, error: %v", i, err0)
+				}
+				wgDone.Done()
+			}()
+
+			cluster := cx.epc
+			member := cluster.Procs[rand.Intn(cluster.Cfg.ClusterSize)]
+			curlReq := e2e.CURLReq{Endpoint: "/v3/watch", Value: string(watchData)}
+
+			args := e2e.CURLPrefixArgs(cluster.Cfg, member, "POST", curlReq)
+			proc, err := e2e.SpawnCmd(args, nil)
+			if err != nil {
+				return fmt.Errorf("failed to spawn: %w", err)
+			}
+			defer proc.Stop()
+
+			// make sure that watch request has been created
+			expectedLine := `"created":true}}`
+			_, lerr := proc.ExpectWithContext(context.TODO(), expectedLine)
+			if lerr != nil {
+				return fmt.Errorf("%v %v (expected %q). Try EXPECT_DEBUG=TRUE", args, lerr, expectedLine)
```

Comment:
We can fail the case directly, no need to return an error.

Reply:
Actually, I handle it in defer. I will update it to make it clear.